chore(max): Capture session IDs in traces

Twixes opened the pull request:
## Problem

We can't link Max traces with session recordings.

## Changes

Adds the session ID to Max traces.

Twixes committed:
chore(max): Capture session IDs in traces

pauldambra committed:
chore: turn on tracing headers

skoob13 commented on ee/hogai/assistant.py:
```diff
@@ -127,6 +128,7 @@ def __init__(
         new_message: Optional[HumanMessage] = None,
         mode: AssistantMode = AssistantMode.ASSISTANT,
         user: User,
+        session_id: str,
```
The Assistant class now runs in different environments, so I'd do like that:
```suggestion
        session_id: str | None = None,
```

I guess the MCP server doesn't really have sessions?

skoob13 reviewed (commented):
Great addition! I struggled today finding a session from the support queue.

pauldambra commented:
tagged in slack but you don't have to do this

if we turn on tracing headers https://github.com/PostHog/posthog/pull/35386 session id and distinct id (will|should) get added to all posthog events automagically

Twixes committed:
Use Paul's suggested approach

https://posthog.slack.com/archives/C02GS3N743W/p1753119247343689?thread_ts=1753119134.473739&cid=C02GS3N743W

Twixes commented:
Applied Paul's better approach to passing the session ID to the backend, and made session_id optional.

Twixes commented on ee/hogai/assistant.py:
```diff
@@ -127,6 +128,7 @@ def __init__(
         user: User,
+        session_id: str,
         contextual_tools: Optional[dict[str, Any]] = None,
         is_new_conversation: bool = False,
```
Ah yeah, the Python LLM observability doesn't expose that neatly. Removed for now, but it _will_ make for a clearer view of traces
